src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs

```diff
@@ -70,6 +74,118 @@ public abstract class QueryableController : ApiController
+					Dictionary<JToken, bool> keyResult = new Dictionary<JToken, bool>();
+
+					foreach (JToken myKey in preMap[mypid])
+					{
+						string keyquoted = JsonConvert.SerializeObject(myKey,
+							new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+						keyResult[myKey] = await proxy.DeleteAsync(collection, keyquoted);
+					}
+
+					parResult[mypid] = keyResult;
+				}
+
+				return Ok(parResult);
+			}
+			catch (Exception e)
+			{
+				return HandleException(e, serviceUri);
+			}
+		}
+
+		protected async Task<IHttpActionResult> AddAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+				bool[] results = new bool[obj.Length];
+				Dictionary<JToken, bool> results1 = new Dictionary<JToken, bool>();
+
+				for (int i = 0; i < obj.Length; i++)
+				{
+					//Serialize the key from the json body and put it into a string.
+					string keyquoted = JsonConvert.SerializeObject(obj[i].Key,
+						new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+					//Serialize the value from the json body and put it into a string.
+					string valuequoted = JsonConvert.SerializeObject(obj[i].Value,
+						new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+					//Fetch the proxy
+					var proxy = await GetServiceProxyForAddAsync<IQueryableService>(serviceUri, obj[i].PartitionId)
+						.ConfigureAwait(false);
+					results[i] = await proxy.AddAsync(collection, keyquoted, valuequoted).ConfigureAwait(false);
+					results1[obj[i].Key] = results[i];
+				}
+				return Ok(results1);
+			}
+			catch (Exception e)
+			{
+				return HandleException(e, serviceUri);
+			}
+		}
+
+		protected async Task<IHttpActionResult> UpdateAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+				bool[][] results = new bool[obj.Length][];
+				Dictionary<JToken, bool[]> results1 = new Dictionary<JToken, bool[]>();
+				for (int i = 0; i < obj.Length; i++)
+				{
+					//Serialize the key from the json body and put it into a string.
+					string keyquoted = JsonConvert.SerializeObject(obj[i].Key,
+						new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+					//Serialize the value from the json body and put it into a string.
+					string valuequoted = JsonConvert.SerializeObject(obj[i].Value,
+						new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+
+					var proxy = await GetServiceProxyForPartitionAsync<IQueryableService>(serviceUri, obj[i].PartitionId)
+						.ConfigureAwait(false);
+					results[i] = await Task.WhenAll(proxy.Select(p => p.UpdateAsync(collection, keyquoted, valuequoted)))
```

Comment:
Can we require partition-id here?  The same way we do for Delete?  If they're updating an entry, they should tell us exactly which one (and only one) they want updated.  Since the "same" key in two different partitions most likely has different meanings - we should not assume we should update them both.